lib/req/steps.ex

```diff
@@ -512,10 +514,15 @@ defmodule Req.Steps do
         :error ->
           cond do
             options = request.options[:connect_options] ->
-              Req.Request.validate_options(options, MapSet.new([:timeout, :protocol]))
+              Req.Request.validate_options(options, MapSet.new([:timeout, :protocol, :proxy]))
 
               pool_opts = [
-                conn_opts: [transport_opts: [timeout: options[:timeout] || 30_000]],
+                conn_opts: [
+                  transport_opts: [
+                    timeout: options[:timeout] || 30_000
+                  ],
+                  proxy: options[:proxy]
```

Comment:
does Mint accept `proxy: nil` or we need to conditionally set the proxy key? Maybe that's why we have test failures?

Reply:
`proxy: nil` seems not to be ok:

```
iex> Mix.install([:mint])
iex> Mint.HTTP.connect(:http, "httpbin.org", 80)
{:ok, %Mint.HTTP1{}}
iex> Mint.HTTP.connect(:http, "httpbin.org", 80, proxy: nil)
** (CaseClauseError) no case clause matching: {:ok, nil}
    (mint 1.4.2) lib/mint/http.ex:359: Mint.HTTP.connect/4
    iex:5: (file)
```

Reply:
Hello, after testing again I figured out that the error in `:timeout` test was indeed caused by `proxy: nil`.
I can't explain why only this test had this issue and why the request appears to work with `proxy: nil`, but I removed this possibility and now all tests are passing.